### PR TITLE
Fix Batch build issues

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by matthchr** [comment]
Actually I think I need to re-enable Batch in the CI build. @shahabhijeet how do I do that?

File: src/SDKs/Batch/DataPlane/Azure.Batch.ProtocolTests/Azure.Batch.ProtocolTests.csproj [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <ItemGroup>
     <PackageReference Include="xunit" Version="2.3.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-preview-20170222-09" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
     <!-- <PackageReference Include="Microsoft.NETCore.App" Version="1.0.2" /> -->
```
Comment (dsgouda) [inline src/SDKs/Batch/DataPlane/Azure.Batch.ProtocolTests/Azure.Batch.ProtocolTests.csproj]: nit: Consider removing commented lines

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great for the most part, please address the comment

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

**Comment by shahabhijeet** [comment]
@matthchr it failed again
19:26:32   c:\workspace\VSUpgradeForTest\src\SDKs\Batch\Support\FileConventions\Tests\AzureBatchFileConventions.Tests\AzureBatchFileConventions.Tests.csproj : error NU1603: Castle.Core 4.0.0 depends on System.ComponentModel.TypeConverter (>= 4.0.1) but System.ComponentModel.TypeConverter 4.0.1 was not found. An approximate best match of System.ComponentModel.TypeConverter 4.1.0 was resolved.

**Comment by dsgouda** [comment]
@matthchr please update the branch update the PR

**Comment by matthchr** [comment]
@dsgouda I believe I am waiting for @shahabhijeet to re-run the tests again after my latest push (commit b6d2ba9)

**Comment by shahabhijeet** [comment]
@matthchr this failed again
https://azuresdkci.westus2.cloudapp.azure.com/view/NetCore-SDK/job/CI-NetSDK_U/100/console

**Comment by shahabhijeet** [comment]
@matthchr at this point, even if this ends up fixing the issue, this PR will not be accepted as it already has merge commits. So a new PR with clean commit history would be required.
We are getting strict on making sure the commit history is clean and we are no longer going to reply on "Squash and Merge" of GitHub. We are seeing increasing amount of merge conflicts when we merge public master to other branches (namely mult-api branch) which is based off the public/master.

**Comment by matthchr** [comment]
You keep adding merge commits - not me.

**Comment by matthchr** [comment]
It looks like the previous error did get resolved though.

Now I have:
```
  Failed   BatchClientIntegrationTests.CloudTaskIntegrationTests.Bug1432996GetTaskOnJobOperationsAndJob
EXEC : error Message:  [c:\workspace\VSUpgradeForTest\build.proj]
   System.AggregateException : One or more errors occurred.
  ---- System.AggregateException : One or more errors occurred.
  -------- System.ArgumentException : Missing required environment variable MABOM_BatchAccountSubscriptionId
  ---- The following constructor parameters did not have matching fixture data: PaasWindowsPoolFixture poolFixture
  Stack Trace:
  
  ----- Inner Stack Trace #1 (System.AggregateException) -----
     at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
     at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
     at System.Threading.Tasks.Task`1.get_Result()
     at BatchClientIntegrationTests.Fixtures.PoolFixture..ctor(String poolId)
     at BatchClientIntegrationTests.Fixtures.PaasWindowsPoolFixture..ctor()
  ----- Inner Stack Trace -----
     at BatchTestCommon.TestCommon.TestConfiguration.GetEnvironmentVariableOrThrow(String environmentSettingName)
     at BatchTestCommon.TestCommon.TestConfiguration..ctor()
     at BatchTestCommon.TestCommon.<>c.<.cctor>b__4_0()
     at System.Lazy`1.CreateValue()
  --- End of stack trace from previous location where exception was thrown ---
     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
     at System.Lazy`1.get_Value()
     at BatchTestCommon.TestCommon.get_Configuration()
     at BatchClientIntegrationTests.IntegrationTestUtilities.TestUtilities.GetCredentialsFromEnvironment()
     at BatchClientIntegrationTests.IntegrationTestUtilities.TestUtilities.<OpenBatchClientFromEnvironmentAsync>d__2.MoveNext()
  ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
  
```

This is because our integration tests don't support recording - you can't run them in CI. Previously they were disabled in CI. Did we lose that configuration?

The following projects should be skipped:
Azure.Batch.IntegrationTests
Azure.Batch.ProtocolTests
AzureBatchFileConventions.IntegrationTests
Batch.FileStaging.Tests

**Review by dsgouda — APPROVED** [review]

**Pull request merged**